python/ray/serve/tests/test_request_timeout.py

```diff
@@ -300,11 +299,116 @@ def get_num_running_tasks():
     results = ray.get([do_request.remote() for _ in range(10)])
     assert all(r.status_code == 408 for r in results)
 
-    # Signal the handlers to exit. After that, no actor tasks should be running
-    # aside from long poll.
-    ray.get(signal_actor.send.remote())
+    # The tasks should all be cancelled.
     wait_for_condition(lambda: get_num_running_tasks() == 0)
 
 
+@pytest.mark.skipif(
```

Comment:
Can you add a test for a timeout while a response is being streamed back to the client?

Reply:
there is one using handle calls: `test_cancel_generator_async`

I'll add an HTTP one quickly as well

Reply:
`test_cancel_generator_async` cancels the request via disconnect. To clarify, I'm suggesting we add a test that cancels the request via timeout.

Reply:
Ah, actually I remember why I didn't add this, it's already covered in `tests/test_streaming_response.py::test_http_disconnect`

Reply:
Oh nice! Do you know if the cancellations during timeout get propagated to downstream deployments as well?

Reply:
yes they do